corehq/apps/reports/standard/deployments.py

```diff
@@ -483,6 +488,40 @@ def _fmt_timestamp(timestamp):
         result[0][1] = table
         return result
 
+    def get_location_column(self, user):
+        assigned_loc_ids = user.get('assigned_location_ids')
+        if not assigned_loc_ids:
+            return '---'
+        primary_loc_id = user.get('location_id')
+        return self._get_formatted_assigned_location_names(primary_loc_id, assigned_loc_ids)
+
+    def _get_formatted_assigned_location_names(self, primary_location_id, assigned_location_ids):
+        """
+        Create an HTML formatted string of the given assigned location names.
+        The primary location will be highlighted in bold.
+        """
+        locs = SQLLocation.objects.filter(location_id__in=assigned_location_ids)
```

Comment:
I think we need some sort of caching here. If reports loads say 100 users at once, we would be firing up 100 queries which could lead to bad performance.

Reply:
We could add a quickcache here, however we will still need to do the initial 100 queries.

I'm thinking just before looping through users [here](https://github.com/dimagi/commcare-hq/blob/1359e6ed5b12d929d97f5a59f0f0181a3811ccdc/corehq/apps/reports/standard/deployments.py#L319), we could do an initial loop through the users to retrieve all their assigned location IDs and map this to a dict. This would look something like this:
```
@memoized
def user_locations_dict(self, users):
    all_loc_ids = []
    for user in users:
        all_loc_ids += user.get('assigned_location_ids')
    all_locs = SQLLocation.objects.filter(location_id__in=all_loc_ids).values_list('location_id', 'name')
    return {
        loc_id: loc_name
        for loc_id, loc_name in all_locs
    }
```
The above causes us to do a few extra loops, however we only need to do a single DB call to get all the location names. This dict can then be passed to the func responsible for parsing the HTML. @mkangia do you think this is a good approach or should using a quickcache be sufficient enough?

Reply:
This seems good to me. One db query sounds good.

You could use a set instead of list to avoid repeated ids.

Also I learnt this recently from Ethan
```
return dict(SQLLocation.objects.filter(
    location_id__in=all_loc_ids
).values_list('location_id', 'name'))
```
will give you the dict you need.

Reply:
Thanks for the code, this is very useful. I've implemented the necessary changes so that we only do a single DB hit to get all user locations for a page ([7ebeb72](https://github.com/dimagi/commcare-hq/pull/34744/commits/7ebeb724ef171090285e0f99695bfa81350528a5)). Unfortunately I couldn't add `memoized` or `quickcache` due to the fact that we are passing a list of users to the func (which is unhashable).